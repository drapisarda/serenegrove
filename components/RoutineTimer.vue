<template>
  <div class="routine-timer">
    <div class="container is-max-desktop">
      {{ formattedTime(currentTime) }}
      <div class="routine-timer__bar" v-if="showBar">
        <div class="routine-timer__bar-fill" :style="remainingTimeStyle"></div>
        <div class="routine-timer__bar-text">
          {{ formattedTime(currentTime) }}
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref, computed, onUnmounted, watch } from 'vue';
import { formattedTime } from '@/composables/formattedTime';

const props = withDefaults(defineProps<{
  time: number,
  start: boolean,
  stop: boolean,
}>(), {
  time: 0,
  start: false,
  stop: false,
});

const showBar = false;

let intervalId: NodeJS.Timer | null = null;
let currentTime = ref(props.time);
let totalTime = ref(props.time);

const remainingTimeStyle = computed(() => {
  return `transform: scaleX(${currentTime.value / totalTime.value});`;
})

const startTimer = () => {
  currentTime.value = props.time;
  intervalId = setInterval(() => {
    if (props.start) currentTime.value -= 1;
  }, 1000);
};

const stopTimer = () => {
  if (!intervalId) return;

  clearInterval(intervalId);
  intervalId = null;
};

watch(() => props.start, (newVal) => {
  if (newVal && !intervalId) startTimer();
});

watch(() => props.stop, (newVal) => {
  if (newVal) {
    stopTimer();
  }
});

watch(() => props.time, (newVal) => {
  stopTimer();
});

onUnmounted(stopTimer);
</script>


<style lang="scss">
@import "@/style/vars.scss";

.routine-timer {
  text-align: center;
  &__bar {
    border: 1px solid $clear-1;
    display: flex;
    justify-content: flex-end;
    height: 2em;
    font-size: 1rem;
    color: $clear-1;
    overflow: hidden;
    position: relative;
  }

  &__bar-text {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    text-align: center;
    font-size: 1em;
    color: $dark-3;
  }

  &__bar-fill {
    height: 100%;
    width: 100%;
    background-color: $clear-5;
    transition: transform 1s ease-in-out;
    transform-origin: right center;
  }
}
</style>