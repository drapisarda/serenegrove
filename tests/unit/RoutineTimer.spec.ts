import RoutineTimer from '@/components/RoutineTimer.vue'
import { mount, type VueWrapper } from '@vue/test-utils'
import { expect } from 'vitest'

describe('RoutineTimer component test', async () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = mount(RoutineTimer, {
      props: {
        start: false,
        time: 400,
      },
    })
  })

  test('RoutineTimer mounts', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
