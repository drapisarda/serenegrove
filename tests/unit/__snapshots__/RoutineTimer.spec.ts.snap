// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoutineTimer component test > RoutineTimer mounts 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="routine-timer"
        start="false"
      >
        <div
          class="container is-max-desktop"
        >
          06:40
        </div>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {
          "pauseTimer": [Function],
          "startTimer": [Function],
          "stopTimer": [Function],
        },
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {
            "pauseTimer": [Function],
            "startTimer": [Function],
            "stopTimer": [Function],
          },
          "_value": {
            "pauseTimer": [Function],
            "startTimer": [Function],
            "stopTimer": [Function],
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "_depsLength": 5,
        "_dirtyLevel": 0,
        "_runnings": 0,
        "_shouldSchedule": false,
        "_trackId": 1,
        "active": true,
        "allowRecurse": true,
        "deps": [
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
        "trigger": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {
          "pauseTimer": [Function],
          "startTimer": [Function],
          "stopTimer": [Function],
        },
      },
      "refs": {
        "VTU_COMPONENT": {
          "pauseTimer": [Function],
          "startTimer": [Function],
          "stopTimer": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "_depsLength": 5,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {
          "pauseTimer": [Function],
          "startTimer": [Function],
          "stopTimer": [Function],
        },
      },
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {
            "start": false,
          },
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "time": 400,
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "currentTime": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": 400,
              "_value": 400,
              "dep": Map {
                ReactiveEffect {
                  "_depsLength": 1,
                  "_dirtyLevel": 0,
                  "_runnings": 0,
                  "_shouldSchedule": false,
                  "_trackId": 1,
                  "active": true,
                  "computed": ComputedRefImpl {
                    "__v_isReadonly": true,
                    "__v_isRef": true,
                    "_cacheable": true,
                    "_setter": [Function],
                    "_value": "06:40",
                    "dep": Map {
                      ReactiveEffect {
                        "_depsLength": 1,
                        "_dirtyLevel": 0,
                        "_runnings": 0,
                        "_shouldSchedule": false,
                        "_trackId": 1,
                        "active": true,
                        "allowRecurse": true,
                        "deps": [
                          [Circular],
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "scheduler": [Function],
                        "trigger": [Function],
                      } => 1,
                    },
                    "effect": [Circular],
                    "getter": [Function],
                  },
                  "deps": [
                    [Circular],
                  ],
                  "fn": [Function],
                  "scheduler": undefined,
                  "trigger": [Function],
                } => 1,
              },
            },
            "displayTime": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_setter": [Function],
              "_value": "06:40",
              "dep": Map {
                ReactiveEffect {
                  "_depsLength": 1,
                  "_dirtyLevel": 0,
                  "_runnings": 0,
                  "_shouldSchedule": false,
                  "_trackId": 1,
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "scheduler": [Function],
                  "trigger": [Function],
                } => 1,
              },
              "effect": ReactiveEffect {
                "_depsLength": 1,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "computed": [Circular],
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "scheduler": undefined,
                "trigger": [Function],
              },
              "getter": [Function],
            },
            "intervalId": null,
            "pauseTimer": [Function],
            "play": false,
            "props": {
              "time": 400,
            },
            "startTimer": [Function],
            "stopTimer": [Function],
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "_depsLength": 1,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "pauseTimer": [Function],
            "startTimer": [Function],
            "stopTimer": [Function],
          },
          "exposed": {
            "pauseTimer": [Function],
            "startTimer": [Function],
            "stopTimer": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "time": 400,
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "time": {
                "0": false,
                "1": true,
                "default": 0,
                "type": [Function],
              },
            },
            [
              "time",
            ],
          ],
          "provides": {},
          "proxy": {
            "time": 400,
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "_depsLength": 1,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_setter": [Function],
                  "_value": "06:40",
                  "dep": Map {
                    ReactiveEffect {
                      "_depsLength": 1,
                      "_dirtyLevel": 0,
                      "_runnings": 0,
                      "_shouldSchedule": false,
                      "_trackId": 1,
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "scheduler": [Function],
                      "trigger": [Function],
                    } => 1,
                  },
                  "effect": [Circular],
                  "getter": [Function],
                },
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "scheduler": undefined,
                "trigger": [Function],
              },
              ReactiveEffect {
                "_depsLength": 1,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
              ReactiveEffect {
                "_depsLength": 1,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {
              "start": false,
            },
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "currentTime": 400,
            "displayTime": "06:40",
            "intervalId": null,
            "pauseTimer": [Function],
            "play": false,
            "props": {
              "time": 400,
            },
            "startTimer": [Function],
            "stopTimer": [Function],
          },
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "ce": undefined,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "06:40",
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="container is-max-desktop"
                >
                  06:40
                </div>,
                "key": null,
                "patchFlag": 1,
                "props": {
                  "class": "container is-max-desktop",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "targetStart": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "06:40",
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="container is-max-desktop"
                >
                  06:40
                </div>,
                "key": null,
                "patchFlag": 1,
                "props": {
                  "class": "container is-max-desktop",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "targetStart": null,
                "transition": null,
                "type": "div",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="routine-timer"
              start="false"
            >
              <div
                class="container is-max-desktop"
              >
                06:40
              </div>
            </div>,
            "key": null,
            "patchFlag": 16,
            "props": {
              "class": "routine-timer",
              "start": false,
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "targetStart": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/danielerapisarda/projects/serenegrove/components/RoutineTimer.vue",
            "__name": "RoutineTimer",
            "components": {},
            "props": {
              "time": {
                "default": 0,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": [
            [Function],
          ],
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="routine-timer"
          start="false"
        >
          <div
            class="container is-max-desktop"
          >
            06:40
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
          "start": false,
          "time": 400,
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "__file": "/Users/danielerapisarda/projects/serenegrove/components/RoutineTimer.vue",
          "__name": "RoutineTimer",
          "components": {},
          "props": {
            "time": {
              "default": 0,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="routine-timer"
          start="false"
        >
          <div
            class="container is-max-desktop"
          >
            06:40
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.4.36",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {
    "pauseTimer": [Function],
    "startTimer": [Function],
    "stopTimer": [Function],
  },
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {
      "pauseTimer": [Function],
      "startTimer": [Function],
      "stopTimer": [Function],
    },
  },
  "wrapperElement": <div
    class="routine-timer"
    start="false"
  >
    <div
      class="container is-max-desktop"
    >
      06:40
    </div>
  </div>,
}
`;
